frontend/src/pages/ViewQuestionPage/ViewQuestion.page.tsx

```diff
@@ -49,52 +55,91 @@ export const qnLoader: LoaderFunction<Question> = async ({ params }) => {
 };
 
 const InnerViewQuestion = () => {
-  const { qn, chat, lang, changeLang, currSubmission, submitCode } =
-    useContext(SharedEditorContext);
+  const {
+    qn,
+    chat,
+    lang,
+    replaceCode,
+    changeLang,
+    currSubmission,
+    submitCode,
+  } = useContext(SharedEditorContext);
+
   return (
     <>
       {qn ? <QnDrawer question={qn} size="xl" /> : <></>}
-      <HStack className="fit-parent" padding={2.5}>
-        <VStack className="fit-parent" gap="2">
-          <Select
-            defaultValue={lang}
-            onChange={(e) => changeLang(e.currentTarget.value as language)}
-          >
-            {LanguageData.map((data) => (
-              <option value={data.lang} key={data.lang}>
-                {data.repr}
-              </option>
-            ))}
-          </Select>
-          <Box width="100%" height="95%">
-            <CollabEditor />
-          </Box>
-        </VStack>
-        <VStack h="100%" w="30%">
-          <Box className="fit-parent">
-            <Center>
-              <QnSubmissionHistory />
-            </Center>
-          </Box>
-          <Box className="fit-parent">
-            <Center>
-              <ChatBox />
-            </Center>
-          </Box>
-          <Box h="10%" w="100%">
-            <Center>
-              <Button
-                w="100%"
-                colorScheme="teal"
-                isLoading={!!currSubmission}
-                onClick={submitCode}
-              >
-                Submit
-              </Button>
-            </Center>
-          </Box>
-        </VStack>
-      </HStack>
+      {!lang ? (
+        <></>
+      ) : (
```

Comment:
From a UX perspective, we might want to pick a default language so it is more intuitive for the user that the editor has loaded

Reply:
for this it would be replaced with a loading block. lang is only null if the initer of the coop session has not loaded in.